kong/api/routes/plugins.lua

```diff
@@ -104,7 +104,7 @@ local function patch_plugin(self, db, _, parent)
       content_type = lower(content_type)
       if find(content_type, "application/x-www-form-urlencoded", 1, true) == 1 or
          find(content_type, "multipart/form-data",               1, true) == 1 then
-        post = arguments.decode(post, kong.db.plugins.schema)
+        post = arguments.decode_and_infer(post, kong.db.plugins.schema)
```

Comment:
Do we still have the old `arguments.decode` around, or is this a full rename of the method?

It's possible that someone else (e.g. a custom plugin) uses this and we inadvertedly break backwards compatibility with this rename (despite this not being part of our pdk)

Reply:
It's a rename.
The old `decode` is now in `arguments_decoder.lua`. Previously that function was doing decoding **and** inferring together, which created tightly bound dependencies on the whole module, so I separated the roles in order to split it in 2 files. `decode_and_infer` is effectively equivalent to the previous `decode` in arguments.lua. We could keep calling it the same as before, to keep it compatible with the previous interface (although that means we will have `arguments.decode` that really does decoding + inferring and `arguments_decoder.decode` that does decoding).

Reply:
We could just rename it back to `decode`, and leave a comment that explains what it does and the difference with the function with the same name in the other module.

Reply:
(done)